packages/anasazi/tpetra/test/LOBPCG/cxx_simple.cpp

```diff
@@ -0,0 +1,193 @@
+// @HEADER
+// *****************************************************************************
+//                 Anasazi: Block Eigensolvers Package
+//
+// Copyright 2004 NTESS and the Anasazi contributors.
+// SPDX-License-Identifier: BSD-3-Clause
+// *****************************************************************************
+// @HEADER
+//
+//  This test is for the internal utilities that are used by the modal analysis solver.
+//
+#include "AnasaziConfigDefs.hpp"
+#include "AnasaziTypes.hpp"
+
+#include "AnasaziTpetraAdapter.hpp"
+#include "AnasaziBasicEigenproblem.hpp"
+#include "AnasaziSimpleLOBPCGSolMgr.hpp"
+#include <Teuchos_CommandLineProcessor.hpp>
+#include <Teuchos_SerialDenseMatrix.hpp>
+
+#include <Tpetra_Core.hpp>
+#include <Tpetra_CrsMatrix.hpp>
+
+#include <ModeLaplace1DQ1.hpp>
+
+using namespace Teuchos;
+
+int main(int argc, char *argv[]) 
+{
+  bool boolret;
+
+  Tpetra::ScopeGuard tpetraScope (&argc,&argv);
+  RCP<const Teuchos::Comm<int> > comm = Tpetra::getDefaultComm ();
+  int MyPID = comm->getRank();
+
+  bool testFailed;
+  bool verbose = false;
+  bool debug = false;
+  bool shortrun = false;
+  std::string which("LM");
+
+  CommandLineProcessor cmdp(false,true);
+  cmdp.setOption("verbose","quiet",&verbose,"Print messages and results.");
+  cmdp.setOption("debug","nodebug",&debug,"Print debugging information.");
+  cmdp.setOption("sort",&which,"Targetted eigenvalues (SM or LM).");
+  cmdp.setOption("shortrun","longrun",&shortrun,"Allow only a small number of iterations.");
+  if (cmdp.parse(argc,argv) != CommandLineProcessor::PARSE_SUCCESSFUL) {
+    return -1;
+  }
+  if (debug) verbose = true;
+
+  typedef double                              ST;
```

Comment:
ditto

Reply:
Test is only meaningful for double-precision due to generalized eigenvalue problem being generated.